dragon-scales/src/app/components/StepViewerComponents/StepViewer.jsx

```diff
@@ -47,107 +56,153 @@ const StyledStepPaperActive = styled(Paper)(({ theme }) => ({
     paddingLeft: theme.spacing(2),
     paddingTop: theme.spacing(1),
     paddingBottom: theme.spacing(1),
+}));
 
-}))
-
-const StyledStepTittleTypography = styled(Typography)(({ theme }) => ({
+const StyledStepTitleTypography = styled(Typography)(({ theme }) => ({
     fontWeight: 'bold',
     fontSize: theme.typography.h6.fontSize,
-}))
+}));
 
 const StyledStepContentBlocksTypography = styled(Typography)(({ theme }) => ({
     fontSize: theme.typography.body1.fontSize,
-}))
+}));
 
 const StyledNewContentBox = styled(Box)(({ theme }) => ({
     display: "flex",
     alignItems: "center",
-    paddingTop: "10px"
-
-}))
-
-
-
-
-export default function StepViewer( { stepEntity, markStepState, reloadTask } ) {
+    paddingTop: "10px",
+}));
 
+function StepViewer({ 
+    stepEntity, 
+    markStepState = () => {}, 
+    reloadTask = () => {}, 
+    parentID = '', 
+    parentName = '', 
+}) {
     const { entitySectionIdRef } = useContext(ExplorerContext);
 
+    // Holds as keys the ids of contentBlocks and as values their corresponding refs
     const [step, setStep] = useState(stepEntity);
     const [isActive, setIsActive] = useState(false);
     const [parsedContentBlocksEnt, setParsedContentBlocksEnt] = useState([]);
     const [reloadEditor, setReloadEditor] = useState(0);
     const [deleteDialogOpen, setDeleteDialogOpen] = useState(false);
-
-
-    // Holds as keys the ids of contentBlocks and as values their corresponding refs
+    const [editDialogOpen, setEditDialogOpen] = useState(false);
+    
     const contentBlocksRefs = useRef({});
     const activeContentBlockRef = useRef(null);
-    const newContentBlockRef = useRef(null);  // Used to keep track of the up-to-date content of tiptap
-    const stepViewerRef = useRef(null); // used to keep track of active state
+    const newContentBlockRef = useRef(null);
+    const stepViewerRef = useRef(null);
     const stepRef = useRef(null);
 
     entitySectionIdRef.current[step.ID] = stepRef;
 
+    // Update step when stepEntity changes
+    useEffect(() => {
+        setStep(stepEntity);
+    }, [stepEntity]);
+
+    // Parse content blocks when step changes
+    useEffect(() => {
+        if (step?.comments) {
+            const parsedBlocks = step.comments.map(comment => JSON.parse(comment));
+            setParsedContentBlocksEnt(parsedBlocks);
+            contentBlocksRefs.current = parsedBlocks.reduce((acc, block) => {
+                acc[block.ID] = block.content[block.content.length - 1];
+                return acc;
+            }, {});
+        }
+    }, [step]);
```

Comment:
This useEffect is the same as the one removed at the bottom but removed the FIXME message? why did you remove the FIXME?